feat: (back) OpenAI 관련 로직 초기설계

lincret opened the pull request:
server/src 하위에 openai 폴더 생성 후 controller와 service 파일 추가.
controller에는 ChatGPT에 메시지를 생성하는 기초 함수 몇 가지가 입력되어 있고,
service는 활용 방향에 대해 고민중인 단계입니다.

lincret committed:
feat: (back) OpenAI 관련 로직 초기설계 (진행중)

server/src 하위에 openai 폴더 생성 후 controller와 service 파일 추가. controller에는
ChatGPT에 메시지를 생성하는 기초 함수 몇 가지가 입력되어 있고, service는 활용 방향에 대해 고민중인 단계입니다.

lincret committed:
fix: (back) fix lock file, removing package-lock.json

package-lock.json 삭제 후 yarn add로 chatgpt 재설치 시도.

anzmin commented:
디렉토리 구조에 문제가 있을 것 같아서 말씀 드립니다. 일반적으로 controller에는 api 즉, `GET /api/user` 등의 end-point를 표현하고 내부 비지니스 로직은 service layer에 구현하는게 일반적입니다. 
그런데 현재 구조에는 controller에 gpt와 통신하는 로직이 존재하고 해당 메서드들은 프론트와의 통신을 위한게 아닌 단순히 gpt와의 통신 메서드로 보이네요.

제 생각에는 feedback이라는 커다란 도메인(controller, service)를 두고 외부에 따로 `openapi.service` 등의 provider로 빼는게 좋아보입니다. feedback controller에는 프론트에서 feedback 관련한 api를 호출하는 end-point를 구현하면 될 것 같습니다.

anzmin commented on server/src/openai/openai.controller.ts:
```diff
@@ -0,0 +1,57 @@
+
+async function sendExampleToChatGPT() {
+	const api = new ChatGPTAPI({
+		apiKey: process.env.OPENAI_API_KEY,
```
nestJS에서 환경변수 사용법은 직접 node 글로벌 변수로 얻는게 아니라 configService를 통해 받아오는 것을 권장합니다. user directory에서 확인 해보시면 좋을 것 같아요

anzmin commented on server/src/app.module.ts:
```diff
@@ -6,14 +6,15 @@ import { envConfig } from '@config';
 		UserModule,
 	],
-	controllers: [AppController],
+	controllers: [AppController, OpenaiController],
```
주로 도메인을 나눌 때는 위의 UserModule처럼 해당 도메인의 module을 만든 후, 해당 모듈을 app.module에 import하는게 더 깔끔해보이네요

anzmin commented on server/src/openai/openai.controller.spec.ts:
```diff
@@ -0,0 +1,18 @@
+	it('should be defined', () => {
+		expect(controller).toBeDefined();
+	});
+});
```
현재 테스트는 고려하고 있지 않아서 *.spec과 같은 파일은 다 삭제하셔도 좋을 것 같습니다.

lincret committed:
feat: (back) openai directory 이름 변경 및 기능 확장

디렉토리 이름 변경: openai -> feedback. 관련 controller 및 module 구조 개선. ChatGPTAPI
환경설정값과 관련된 상수 별도 관리. 피드백 관련 DTO 구조 생성중.
